samtranslator/model/sam_resources.py

```diff
@@ -2081,8 +2088,9 @@ def _construct_sqs_queue_policy(
         queue_policy_name = (
             f"{self.logical_id}QueuePolicyDestination{dest_index}" if multi_dest else f"{self.logical_id}QueuePolicy"
         )
+        # For multi-destination connector, modifying resource attribute will affect others, use deep copy to prevent it
```

Comment:
> For multi-destination connector, modifying resource attribute will affect others

Why?

Reply:
When we use multi-destination and embedded connector together 
```
MyServerlessFunction:
    Type: AWS::Serverless::Function
    Connectors:
      MixedDestConnector:
        Properties:
          Metadata: xxx
          Destination:
          - Id: MyBucket
          - Id: MyQueue
          Permissions:
          - Read
          - Write
         ...
```
This is going to become two IAM Managed Policies. After generating the first IAM Managed Policy but before the second one, if I modify Metadata, it will it to change for the second IAM Managed Policy generation.

Reply:
Can it be avoided altogether by refactoring code?